lib/model/autocomplete.dart

```diff
@@ -339,6 +432,12 @@ class AutocompleteDataCache {
   void invalidateUser(int userId) {
     _nameWordsByUser.remove(userId);
   }
+
+  final Map<String, String> _namesLowercased = {};
+
+  String nameLowercased(String name) {
+    return _namesLowercased[name] ??= name.toLowerCase();
+  }
```

Comment:
Is there a reason not to key this by user ID, like `_nameWordsByUser` is? (If doing that, we should make sure `invalidateUser` acts appropriately on the lowercased-name data.)

Reply:
How about `_lowercasedNameByUser` and `lowercasedNameForUser`, on the pattern of `_nameWordsByUser` and `nameWordsForUser`?

Reply:
> Is there a reason not to key this by user ID, like `_nameWordsByUser` is?

Two reasons: a) to not pass the whole `User` object while we can do the work with just one of its properties. b) there can be multiple users with the same name so we can save the space a little bit by having just one key-value pair for all those user names.
However, this will bring some complexity when adding the appropriate logic in `invalidateUser`. Let's assume we somehow pass `fullName` to 'invalidateUser' besides `userId` to remove the lowercase name. It will remove the user name for all the users that share the same name. 
So I am not sure whether to key it by `userId` or `fullName`?

Reply:
Yeah, I think best to key it by user ID, the same way as `_nameWordsByUser`.

It's uncommon for users to have the same name, so there's minimal space savings to be had from that, which makes it not worth the complexity of worrying about interactions there.

Relatedly: in `nameWordsForUser` we're calling `toLowerCase` on `user.fullName`, just like this does. So let's have that step share this same cache.